src/main/java/mate/academy/bookstore/mapper/BookMapper.java

```diff
@@ -1,14 +1,37 @@
 package mate.academy.bookstore.mapper;
 
+import java.util.Set;
+import java.util.stream.Collectors;
 import mate.academy.bookstore.config.MapperConfig;
 import mate.academy.bookstore.dto.book.BookDto;
+import mate.academy.bookstore.dto.book.BookDtoWithoutCategoryIds;
 import mate.academy.bookstore.dto.book.CreateBookRequestDto;
 import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.model.Category;
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
+import org.mapstruct.MappingTarget;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
     BookDto toDto(Book book);
 
     Book toModel(CreateBookRequestDto requestDto);
+
+    BookDtoWithoutCategoryIds toDtoWithoutCategories(Book book);
+
+    @AfterMapping
+    default void setCategoryIds(@MappingTarget BookDto bookDto, Book book) {
+        Set<Long> categoryIds = book.getCategories().stream()
+                .map(Category::getId)
+                .collect(Collectors.toSet());
+        bookDto.setCategoryIds(categoryIds);
+    }
+
+    @AfterMapping
+    default void setCategories(@MappingTarget Book book, CreateBookRequestDto requestDto) {
+        book.setCategories(requestDto.getCategoryIds().stream()
+                .map(Category::new)
+                .collect(Collectors.toSet()));
+    }

```

Comment:
I am not sure it's the correct functionality to create new Categories

Reply:
![image](https://github.com/LaverAndrii/Book-Store/assets/119442607/7cc98955-e364-441d-ac30-ce64bf9fa9bf)
Similar method in Spring Boot Web: Mappers in Details theme